src/memcpy_vfp.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Dmitry Osipenko
+ * Copyright (c) Erik Faye-Lund
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef __TEGRA_MEMCPY_H
+#define __TEGRA_MEMCPY_H
+
+#include <stdbool.h>
+
+void tegra_copy_block_vfp(char *dst, char *src, int size,
+                          bool bounce, bool move);
+
+void tegra_do_memcpy_vfp(void *dst, const void *src, int size, bool bounce);
+
+static inline void
+tegra_memcpy_vfp(void *dst, const void *src, int size,
+                 bool src_cached, bool dst_cached)
+{
+    if (__builtin_expect(dst == src || !size, 0))
+        return;
+
+    tegra_do_memcpy_vfp(dst, src, size, !(src_cached || dst_cached));
```

Comment:
This only gets called with `src_cached = FALSE`, `dst_cached = TRUE`, which leads to `bounce` always being false.

Reply:
Nothing stops to get different calls sometime later.

Reply:
Sure, I was just trying to reason about the code-paths that end up here...